src/DDTrace/Configuration/EnvVariableRegistry.php

```diff
@@ -18,21 +18,52 @@ public function __construct()
         $this->registry = [];
     }
 
+    /**
+     * Return an env variable that starts with "DD_".
+     *
+     * @param string $key
+     * @return string|null
+     */
+    public static function get($key)
+    {
+        $value = getenv(self::convertKeyToEnvVariableName($key));
+        if (false === $value) {
+            return null;
+        }
+        return trim($value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stringValue($key, $default)
+    {
+        return isset($this->registry[$key])
+            ? $this->registry[$key]
+            : $this->registry[$key] = (string) self::get($key);
+    }
+
     /**
      * {@inheritdoc}
      */
     public function boolValue($key, $default)
     {
-        if (!isset($this->registry[$key])) {
-            $value = getenv($this->convertKeyToEnvVariableName($key));
-            $value = trim(strtolower($value));
-            if ($value === '1' || $value === 'true') {
-                $this->registry[$key] = true;
-            } elseif ($value === '0' || $value === 'false') {
-                $this->registry[$key] = false;
-            } else {
-                $this->registry[$key] = $default;
-            }
+        if (isset($this->registry[$key])) {
+            return $this->registry[$key];
+        }
+
+        $value = self::get($key);
+        if (null === $value) {
+            return $default;
```

Comment:
[non blocking] Shouldn't we store the result in `$this->registry[$key]` and remember for the next time? Keep in mind that this is never supposed to be called by users (which may provide different defaults), it is always called by a configuration instance. We may even considering to protect somehow this class.

Reply:
I can totally change it back to that. My thought was that saving the fallback value in the registry  causes this behavior:

```php
var_dump($config->getString('some_unset_val', 'foo')); // "foo"
// ...and then later on...
var_dump($config->getString('some_unset_val', 'bar')); // "foo" (still)
```

I feel like this could be the cause of esoteric bugs later on. :D But if that's the desired behavior, I'm totally down for changing it back!